### PR TITLE
feat: Introduce security integrations datasource

### PR DESCRIPTION
**Pull request by sfc-gh-asawicki**
Added security integrations datasource, for all types of security integrations:
- new datasource implemented
- new datasource initially tested
- examples added
- docs generated

Additional changes:
- slightly improved show output schemas generator to generate schema for any other SDK object (with some todos for later)
- fixed previous datasources descriptions
- fixed migration guide entry

What is missing:
- describe schema is not created (waiting for all security integrations PRs)
- testing all integrations (waiting for all security integrations PRs)

### CODE REVIEW

File: docs/data-sources/databases.md [hunk 1]
```diff
@@ -2,12 +2,12 @@
 page_title: "snowflake_databases Data Source - terraform-provider-snowflake"
 subcategory: ""
 description: |-
-  Datasource used to get details of filtered databases. Filtering is aligned with the current possibilities for SHOW DATABASES (https://docs.snowflake.com/en/sql-reference/sql/show-databases) query (`like`, 'starts_with', and `limit` are all supported. The results of SHOW, DESCRIBE, and SHOW PARAMETERS IN are encapsulated in one output collection.
+  Datasource used to get details of filtered databases. Filtering is aligned with the current possibilities for SHOW DATABASES https://docs.snowflake.com/en/sql-reference/sql/show-databases query (like, 'starts_with', and limit are all supported). The results of SHOW, DESCRIBE, and SHOW PARAMETERS IN are encapsulated in one output collection.
```
Comment (sfc-gh-jcieslak) [inline docs/data-sources/databases.md:5]: like + limit have different quoting than starts_with

**Review by sfc-gh-jcieslak — DISMISSED** [review]

File: docs/data-sources/databases.md (hunk 1 above)
Comment (sfc-gh-asawicki) [reply docs/data-sources/databases.md:5]: 🤦  fixed.

**Review by sfc-gh-jcieslak — APPROVED** [review]

File: pkg/schemas/security_integration_property_gen.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+package schemas
+
+import (
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+// ShowSecurityIntegrationPropertySchema represents output of SHOW query for the single SecurityIntegrationProperty.
+var ShowSecurityIntegrationPropertySchema = map[string]*schema.Schema{
```
Comment (sfc-gh-jmichalak) [inline pkg/schemas/security_integration_property_gen.go:11]: Not Show, but Describe. Change in other places as well.

File: pkg/schemas/security_integration_property_gen.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+	},
+	"default": {
+		Type:     schema.TypeString,
+		Computed: true,
+	},
+}
+
+var _ = ShowSecurityIntegrationPropertySchema
+
+func SecurityIntegrationPropertyToSchema(securityIntegrationProperty *sdk.SecurityIntegrationProperty) map[string]any {
```
Comment (sfc-gh-jmichalak) [inline pkg/schemas/security_integration_property_gen.go:32]: Why passing a pointer?

**Review by sfc-gh-jmichalak — CHANGES_REQUESTED** [review]

File: pkg/schemas/security_integration_property_gen.go (hunk 3 above)
Comment (sfc-gh-asawicki) [reply pkg/schemas/security_integration_property_gen.go:32]: This is generated and it's everywhere a pointer, which is basically a bit faster. So why not a pointer?

File: pkg/schemas/security_integration_property_gen.go (hunk 2 above)
Comment (sfc-gh-asawicki) [reply pkg/schemas/security_integration_property_gen.go:11]: It's explained in the generator comment

**Review by sfc-gh-jmichalak — APPROVED** [review]

**Pull request merged**